packages/kit/src/utils/url.js

```diff
@@ -75,6 +75,21 @@ export function decode_params(params) {
 	return params;
 }
 
+/**
+ * The error when a URL is malformed is not very helpful, so we augment it with the URI
+ * @param {string} uri
+ */
+export function decode_uri(uri) {
+	try {
+		return decodeURI(uri);
+	} catch (e) {
+		if (e instanceof Error) {
+			e.message = `Failed to decode URI: ${uri}\n` + e.message;
```

Comment:
I maybe would leave off the old message here since it doesn't add anything useful (`URIError: URI malformed`)

Reply:
Maybe (?) there's another kind of error that can be thrown, so I was playing it safe and just prepending our own.